### ISSUE TITLE
Alert close button names are too generic

### ISSUE THREAD

**Issue by bdryanovski** [clr-team/planning#64]
## Description of work

The close button on an alert can be too generic in cases where there are multiple close buttons on the page (like inside a modal or with more than one alert open), and screen readers should have better details read out.

## Issue(s)

* https://github.com/vmware/clarity/issues/3498

## Definition of done

* [X] Tests completed
* [ ] Reviewed by peers
* [ ] Design sign off (if necessary)
* [ ] Merged into master/topic branch
* [ ] Backport (if necessary)
